src/main/java/net/dries007/tfc/common/recipes/BarrelRecipeWrapper.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.recipes;
+
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import net.dries007.tfc.common.recipes.inventory.EmptyInventory;
+
+public interface BarrelRecipeWrapper extends IItemHandlerModifiable, IFluidHandler, EmptyInventory
```

Comment:
This should extend `Container`, not `EmptyInventory`, since the latter implies functionality, not an interface contract. It should also be named `BarrelInventory` and in the package `recipes.inventory` (with all the other inventory interfaces). It can be implemented explicitly via FQN where needed.

Reply:
Oop good point on the `Container` not `EmptyInventory`

Russian said I should put it with the recipes & name it `BarrelRecipeWrapper`. Shall make the changes

Reply:
Yeah, sorry, I saw that after seeing this PR, `RecipeWrapper` is Forge's abstraction for `Container` wrapping `IItemHandlerModifiable`, but we use a different naming scheme for our wrappers/inventories which was missed, which is understandable.